spark/src/main/scala/org/apache/spark/sql/comet/execution/shuffle/CometShuffleExchangeExec.scala

```diff
@@ -454,12 +458,13 @@ class CometShuffleWriteProcessor(
     new SQLShuffleWriteMetricsReporter(context.taskMetrics().shuffleWriteMetrics, metrics)
   }
 
+  // FIXME: we need actually prev of rdd?
   override def write(
-      rdd: RDD[_],
+      inputs: Iterator[_],
```

Comment:
If the input iterator has partitionId, we still need to get rid of it. I don't run this but looks like this will cause failures.

Reply:
We have `ShimCometShuffleWriteProcessor.write()` for Spark 3.x

Reply:
Yea, I mean for Spark 4.0 case, we also need to get rid of the partition. I saw you replied my other comment: https://github.com/apache/datafusion-comet/pull/407#discussion_r1602237566

Reply:
I implemented the partitionId removal.
Thanks @viirya